onboarding/main.go

```diff
@@ -155,20 +151,15 @@ func convertRsaPublicKeyAsPemStr(publicKey *rsa.PublicKey) string {
 	return string(publicKeyPem)
 }
 
-func getStorageClusterMetadata(ctx context.Context, operatorNamespace string, clientset *kubernetes.Clientset) (*metav1.PartialObjectMetadata, error) {
-	var storageClusterMetadata metav1.PartialObjectMetadata
-	storageClusterGVKPath := fmt.Sprintf(
-		"/apis/ocs.openshift.io/v1/namespaces/%s/storageclusters/%s",
-		operatorNamespace,
-		storageClusterName,
-	)
-	storageClusterMetadataJSON, err := clientset.RESTClient().Get().AbsPath(storageClusterGVKPath).Do(ctx).Raw()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get storage cluster metadata: %v", err)
+func getStorageClusterMetadata(ctx context.Context, operatorNamespace string, client client.Client) (*metav1.PartialObjectMetadata, error) {
+	storageClusterMetadata := metav1.PartialObjectMetadata{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "storagecluster",
+			APIVersion: "ocs.openshift.io/v1",
+		},
 	}
-
-	if err = json.Unmarshal(storageClusterMetadataJSON, &storageClusterMetadata); err != nil {
-		return nil, fmt.Errorf("failed to parse storage cluster metadata response: %v", err)
+	if err := client.Get(ctx, types.NamespacedName{Name: storageClusterName, Namespace: operatorNamespace}, &storageClusterMetadata); err != nil {
+		return nil, err
```

Comment:
You are now using a client that can handle custom resources. Why can we just get the entire StorageCluster CR into StorageCluster struct? 

It will simplify the code and will make this entire function redundant